charms/kubernetes_snaps.py

```diff
@@ -478,6 +478,12 @@ def configure_scheduler(extra_args_config, kubeconfig):
     )
 
 
+def _enable_config_hashing() -> bool:
+    """This is a debug option to enable config hashing."""
+    env = os.environ.get("JUJU_FEATURE_KUBERNETES_SNAP_CONFIG_HASHING") or ""
+    return env.lower() == "on"
+
+
```

Comment:
This allows the charm at runtime (per hook even!!!) decide whether or not it wants to enable hashing. 

Now an upgrade action could restart the service even if there aren't any config changes.  We could even expose via charm config (though i think that'd be going too far).